ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/pull-queries-with-range-scan.json

```diff
@@ -0,0 +1,354 @@
+        "CREATE TABLE MATVIEW AS SELECT ID, V0, V1 FROM INPUT;",
+        "SELECT ID, V0, V1 FROM MATVIEW WHERE ID >= 12;"
+      ],
+      "properties": {
+        "ksql.query.pull.table.scan.enabled":  false,
+        "ksql.query.pull.range.scan.enabled":  true
+      },
+      "expectedError": {
+        "message": "Query requires table scan to enabled"
+      }
+    },
+    {
+      "name": "range scan on non-strings should fall back to table-scan if enabled",
+      "statements": [
+        "CREATE TABLE INPUT (ID INTEGER PRIMARY KEY, V0 STRING, V1 STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE TABLE MATVIEW AS SELECT ID, V0, V1 FROM INPUT;",
+        "SELECT ID, V0, V1 FROM MATVIEW WHERE ID >= 12;"
+      ],
+      "properties": {
+        "ksql.query.pull.table.scan.enabled":  true,
+        "ksql.query.pull.range.scan.enabled":  true
+      },
+      "inputs": [
+        {"topic": "test_topic", "timestamp": 12345, "key": 10, "value": {"v0": "v10a", "v1": "v10b"}},
+        {"topic": "test_topic", "timestamp": 12355, "key": 11, "value": {"v0": "v11a", "v1": "v11b"}},
+        {"topic": "test_topic", "timestamp": 12365, "key": 12, "value": {"v0": "v12a", "v1": "v12b"}},
+        {"topic": "test_topic", "timestamp": 12375, "key": 13, "value": {"v0": "v13a", "v1": "v13b"}},
+        {"topic": "test_topic", "timestamp": 12385, "key": 14, "value": {"v0": "v14a", "v1": "v14b"}}
+      ],
+      "responses": [
+        {"admin": {"@type": "currentStatus"}},
+        {"admin": {"@type": "currentStatus"}},
+        {"query": [
+          {"header":{"schema":"`ID` INTEGER KEY, `V0` STRING, `V1` STRING"}},
+          {"row":{"columns":[13, "v13a", "v13b"]}},
+          {"row":{"columns":[12, "v12a", "v12b"]}},
+          {"row":{"columns":[14, "v14a", "v14b"]}}
+        ]}
+      ]
+    },
+    {
+      "name": "range scan query must fail for multiple range expressions and table scan disabled",
+      "statements": [
+        "CREATE TABLE INPUT (ID STRING PRIMARY KEY, V0 STRING, V1 STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE TABLE MATVIEW AS SELECT ID, V0, V1 FROM INPUT;",
+        "SELECT ID, V0, V1 FROM MATVIEW WHERE ID >= '11' AND ID <= '13';"
+      ],
+      "properties": {
+        "ksql.query.pull.table.scan.enabled":  false,
+        "ksql.query.pull.range.scan.enabled":  true
+      },
+      "expectedError": {
+        "message": "A comparison condition on the key column cannot be combined with other comparisons such as an IN predicate"
+      }
+    },
+    {
+      "name": "range scan query must fall back to table scan for multiple range expressions if table scan is enabled",
+      "statements": [
+        "CREATE TABLE INPUT (ID STRING PRIMARY KEY, V0 STRING, V1 STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE TABLE MATVIEW AS SELECT ID, V0, V1 FROM INPUT;",
+        "SELECT ID, V0, V1 FROM MATVIEW WHERE ID >= '11' AND ID <= '13';"
```

Comment:
This is one great case to test, one that was mentioned.

I was also trying to ask for testing of `ID >= '11' OR ID <= '13'`.  We'd also like this to be a table scan.